src/Utilities.h

```diff
@@ -28,6 +38,40 @@ double sse_calc(const std::vector<T>& simval, const std::vector<T>& obsval){
 	
 }
 
+/**
+ * @brief Create new vector summating every Nth element
+ * 
+ * Summate values at every Nth index, e.g. n = 7 would summate
+ * daily data to be weekly data
+ * 
+ * @param data_vector	Vector of data to be reduced
+ * @param n	Nth value at which summation should occur (i.e. index+1)
+ * 
+ * @return Vector of summation values
+ */
+template<typename T>
+std::vector<T> AccumulateEveryNth(const std::vector<T>& data_vector, const int& n)
```

Comment:
There's no point in declaring the parameter `n` as `const int&`. This is typically only useful in two cases:

1) To avoid copying a large data structure (vector, string etc.)
2) To provide a guarantee to the caller that a piece of data passed by reference will not be altered by the function

For standard data types like ints, floats etc. (1) is irrelevant, since the cost of passing a reference is no less than passing a copy of the original data. And in light of (1) , (2) is also irrelevant, since it doesn't matter to the caller if the function modifies a local copy of a variable: it doesn't affect the caller.

Just stick with declaring it as `int n`.